anvio/cazymes.py

```diff
@@ -0,0 +1,252 @@
+                              "the safest way to handle things." % (self.cazyme_data_dir, self.cazyme_data_dir))
+
+        if not self.cazyme_data_dir:
+            self.cazyme_data_dir = os.path.join(os.path.dirname(anvio.__file__), 'data/misc/CAZyme')
+
+        filesnpaths.is_output_dir_writable(os.path.dirname(os.path.abspath(self.cazyme_data_dir)))
+
+        self.resolve_database_url()
+
+        if not args.reset and not anvio.DEBUG:
+            self.is_database_exists()
+
+        if args.reset:
+            filesnpaths.gen_output_directory(self.cazyme_data_dir, delete_if_exists=True, dont_warn=True)
+        else:
+            filesnpaths.gen_output_directory(self.cazyme_data_dir)
+
+    def resolve_database_url(self):
+        """Create path to CAZyme ftp
+
+        Added self values
+        ================= 
+        - self.page_index : string
+            version of CAZyme database
+
+        """
+        if self.args.cazyme_version:
+            self.page_index = self.args.cazyme_version 
+            self.run.info('Attempting to use version', self.args.cazyme_version)
+        else:
+            self.page_index = 'V11'
+            self.run.info_single('No CAZyme version specified. Using current release.')
+
+        self.database_url = os.path.join("https://bcb.unl.edu/dbCAN2/download/Databases", f"{self.page_index}", f"dbCAN-HMMdb-{self.page_index}.txt") 
+
+    def is_database_exists(self):
+        """Determine if CAZyme database has already been downloaded"""
+        if os.path.exists(os.path.join(self.cazyme_data_dir, f"dbCAN-HMMdb-{self.page_index}.txt")):
+            raise ConfigError(f"It seems you already have CAZyme database installed in {self.cazyme_data_dir}, please use --reset flag if you want to re-download it.")
+
+    def download(self, hmmpress_files=True):
+        """Download CAZyme database and compress with hmmpress"""
+        self.run.info("Database URL", self.database_url)
+
+        utils.download_file(self.database_url, os.path.join(self.cazyme_data_dir, os.path.basename(self.database_url)) , progress=self.progress, run=self.run)
+
+        if hmmpress_files:
+            self.hmmpress_files()
+
+    def hmmpress_files(self):
+        """Runs hmmpress on CAZyme HMM profiles."""
+
+        file_path = os.path.join(self.cazyme_data_dir, os.path.basename(self.database_url))
+        cmd_line = ['hmmpress', file_path]
+        log_file_path = os.path.join(self.cazyme_data_dir, '00_hmmpress_log.txt')
+        ret_val = utils.run_command(cmd_line, log_file_path)
+
+        if ret_val:
+            raise ConfigError("Hmm. There was an error while running `hmmpress` on the Pfam HMM profiles. "
+                                "Check out the log file ('%s') to see what went wrong." % (log_file_path))
+        else:
+            # getting rid of the log file because hmmpress was successful
+            os.remove(log_file_path)
+
+class CAZyme(object):
```

Comment:
We don't have a citation statement for Pfams either, but I think it would be great to add something similar to our citation statements for other classes/programs so people would remember or see it in their logs that they should CAZymes in their methods sections etc.

Reply:
`THIRD_PARTY_PROGRAMS` are only relevant for workflows, so it wouldn't be necessary unless it will appear in a workflow.